api/src/main/java/org/pmiops/workbench/calhoun/CalhounRetryHandler.java

```diff
@@ -51,14 +50,16 @@ protected void logNoRetry(Throwable t, int responseCode) {
   }
 
   @Autowired
-  public CalhounRetryHandler(
-      BackOffPolicy backoffPolicy, Provider<TermsOfServiceApi> termsOfServiceApiProvider) {
-    super(backoffPolicy, new CalhounRetryPolicy(), termsOfServiceApiProvider);
+  public CalhounRetryHandler(BackOffPolicy backoffPolicy, FireCloudService fireCloudService) {
+    super(
+        backoffPolicy,
+        new CalhounRetryPolicy(),
+        fireCloudService,
+        ExceptionUtils::convertCalhounException);
   }
 
   @Override
   protected WorkbenchException convertException(ApiException exception) {
-    return maybeConvertMessageForTos(exception.getCode())
-        .orElseGet(() -> ExceptionUtils.convertCalhounException(exception));
+    return convertTerraException(exception, exception.getCode());
```

Comment:
I wanted to push the `getCode()` call into `convertTerraException` but the common base class here is simply `Exception` which does not have `getCode()`, so I couldn't find a way to do that.  Solve that problem, and this duplicated code goes away entirely.